test/test_transformers.py

```diff
@@ -1302,6 +1359,57 @@ def test_sdp_fused_grad_against_math(self, contiguous_inputs: bool):
         # Cast up and compare
         self.assertEqual(qkv.grad, qkv_lp.grad.to(torch.float64), atol=1e-5, rtol=1e-5)
 
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not SM80OrLater, "Flash Attention was not built for this system")
```

Comment:
Why not use an OpInfo for this?

Reply:
Cargo culting.. Is there good documentation of another way this and maybe test_nestedtensor.py could be re-written?

Reply:
Maybe I am confused. I do have an opinfo entry for SDPA: https://github.com/pytorch/pytorch/blob/2fc73622f8e832179e275239506d749dd30767b3/torch/testing/_internal/common_methods_invocations.py#L12591

Its just that this test doesn't exactly conform to any of the standard tests, AFAIK. Would instead I add these test somewhere else?

Reply:
scaled_dot_product_attention can be added as an OpInfo by following similar operations, but for NestedTensors we'd need to extend OpInfo to support generating NestedTensor inputs and such. @drisspg, I can point you to the right spots.

Reply:
Can you just make your custom sample function generate NestedTensor? Or there are sizes-like calls in there that break things?

Reply:
Yeah the reason for not adding NestedTensor was because we need to do some work to make the sampling functions work with nested inputs.

I am more curious if it is an anit-pattern to have these one off tests like the one I wrote here if the function is also tested via OpInfo. Should I have written a opfinfo test that uses the machinery but only ends up applying to just that function?

Reply:
The benefit of the opinfo is that this function will get tested for everything (autograd, functorch, pt2, devices correctness, etc).
I agree that if you already have a version there for Tensor inputs it's most of the work.

But as soon as you start to write manual autograd testing for NestedTensor, I feel like fixing up OpInfo is much better, because then you'll get all these other kinds of testing for free!
So my personal opinion is that, unless you're have time constraint to land this, I think you want to fixup OpInfo instead of adding a one-off test.